### ISSUE TITLE
Misattributed hole with intersection

### ISSUE THREAD

**Issue by davebrent** [w8r/martinez#96]
Ive put together a little gist of the issue [here](https://gist.github.com/davebrent/28a686e7555f0e65f0859a6994bce5c0). It seems that the hole in this case is being assigned to the split polygon above rather than the polygon it belonged to.

I think this may be similar to #38?

**The problem**
![actual](https://user-images.githubusercontent.com/1835547/50606308-3908da00-0ebd-11e9-9eec-7f6ed4a9e029.png)

**Expected result**
![expected](https://user-images.githubusercontent.com/1835547/50606310-3908da00-0ebd-11e9-914b-7fe4130e45ec.png)

**Comment by jakeNiemiec** [comment]
I think when polygons have holes, you need to use the MultiPolygon format http://geojson.org/geojson-spec.html#id7 instead of polygon: https://gist.github.com/davebrent/28a686e7555f0e65f0859a6994bce5c0#file-index-js-L21

In my usage, martinez.intersection should return a MultiPolygon in this case.

**Comment by davebrent** [comment]
> I think when polygons have holes, you need to use the MultiPolygon format http://geojson.org/geojson-spec.html#id7 instead of polygon: https://gist.github.com/davebrent/28a686e7555f0e65f0859a6994bce5c0#file-index-js-L21
> 
> In my usage, martinez.intersection should return a MultiPolygon in this case.

Hi Jake thanks but that's not my understanding. A `MultiPolygon` is just an array of `Polygon` which in turn is an array of rings, the first being the exterior the rest being the interior holes. [source](https://tools.ietf.org/html/rfc7946#section-3.1.6)

What I think is happening here is that the ring representing the hole in one polygon is being assigned to a different one output

**Comment by jakeNiemiec** [comment]
>A MultiPolygon is just an array of Polygon coordinate arrays, which are arrays of rings, the first being the exterior the rest being the interior holes.

Right, my point was that you will [always get a `MultiPolygon` as an output](https://github.com/mfogel/polygon-clipping#output), but as far as I can tell that does not influence the original problem 🙃.

> What I think is happening here is that the ring representing the hole in one polygon is being assigned to a different one on output

You are right, the intersection is dividing the input (a poly with a hole) into 2. You can see from [the `expected.json`->`actual.json` diff](https://www.diffchecker.com/G522QWdZ) that the hole is indeed misattributed to the poly furthest from the origin. (or the last poly in a multipoly?)

![image](https://user-images.githubusercontent.com/16782063/50611930-d9e79d00-0e9d-11e9-8ab2-e1a4a548fe1d.png)

I made a similar gist with the example from https://github.com/w8r/martinez/issues/38 by @joelgallant: https://gist.github.com/jakeNiemiec/7d75093e5e8d9fe52008871c109bb855#file-actual-json (I was unable to get [the codepen example](https://github.com/w8r/martinez/issues/38#issuecomment-360522910) by @w8r to load)

This example shows the same problem, but with **martinez.diff**:

![image](https://user-images.githubusercontent.com/16782063/50616758-7b2b1f00-0eaf-11e9-8395-556bc22a9039.png)

With this, you can see that the holes are, again, misattributed to the poly that is the furthest from the origin. These problems seem reflected [in another popular issue](https://github.com/w8r/martinez/issues/69) by @erikh2000 from nearly a year ago.

**TL;DR**: When a subject is divided into 2 polys via diff or intersection clipping, **all of the subjects holes seem to be misattributed to the poly furthest from the origin**. (or the last poly in a multipoly?)

**Comment by rowanwins** [comment]
Think this is all resolved now from #113

**Comment by w8r** [comment]
https://codesandbox.io/s/martinez-playground-ejy9wf
